scan-results.js

```diff
@@ -35,8 +35,13 @@ const scanResult = async(req, selfScan=false) => {
       scannedEmailId = req.body.scannedEmailId;
     }
 
-    if (!selfScan && signedInUser && sha1(signedInUser.email) === req.body.emailHash) {
-      selfScan = true;
+    if (signedInUser) {
+      for (const emailAddress of signedInUser.email_addresses) {
+        if (!selfScan && sha1(emailAddress.email) === req.body.emailHash) {
+          selfScan = true;
+          break;
+        }
+      }
```

Comment:
Could possibly use [`Array.some()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some), but this looks good.

```js
// maybe?
if (signedInUser) {
  selfScan = signedInUser.email_addresses.some(emailAddress => sha1(emailAddress.email) === req.body.emailHash);
}
```